File: backend/app/schemas/character.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class CharacterCreateRequest(BaseModel):
    name: str


class CharacterStatsResponse(BaseModel):
    hp: int
    mp: int
    strength: int
    intelligence: int
    agility: int
    luck: int
    charm: int


class CharacterResponse(BaseModel):
    id: str
    name: str
    race: str
    stats: CharacterStatsResponse
    skills: list
    equipment: dict
    pets: list
    relationships: list
    rarity_score: float
    worldline_count: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: ios/AnotherWorldline/Models/Character.swift
import Foundation

struct GameCharacter: Codable, Identifiable {
    let id: String
    let name: String
    let race: String
    let stats: CharacterStats
    let skills: [String]
    let equipment: [String: String]
    let pets: [String]
    let relationships: [String]
    let rarityScore: Double
    let worldlineCount: Int
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, name, race, stats, skills, equipment, pets, relationships
        case rarityScore = "rarity_score"
        case worldlineCount = "worldline_count"
        case createdAt = "created_at"
    }
}

struct CharacterStats: Codable {
    let hp: Int
    let mp: Int
    let strength: Int
    let intelligence: Int
    let agility: Int
    let luck: Int
    let charm: Int
}
